Методы работы с группой

teodoroven opened the pull request.

teodoroven committed:
Реализован класс IParticipantGroupManager

teodoroven committed:
Добавил тесты

Kemsekov reviewed (commented):
`User GroupOwner`

При выходе из конструктора имеет значение null.

Почему бы прямо в конструкторе не найти владельца группы?

Kemsekov commented on DeadlineNetwork/Server/App/Controllers/IParticipantGroupManager.cs:
```diff
@@ -0,0 +1,135 @@
+    public Group Group { get; }
+    public User GroupOwner { get; }
+
+    public ParticipantGroupManager(ApplicationDbContext db, User participant, Group group)
```
Почему нельзя сразу найти значение GroupOwner?
У тебя при выходе из конструктора значение null в этом параметре.

Сделай запрос в бд да найди его

teodoroven committed:
Исправлен GroupOwner

teodoroven committed:
Исправил тесты

Kemsekov commented on DeadlineNetwork/Server/App/Controllers/IParticipantGroupManager.cs:
```diff
@@ -0,0 +1,136 @@
+    
+}
+
+public class ParticipantGroupManager : IParticipantGroupManager
```
ВЫнести в отдельный файл ParticipantGroupManager.cs

Kemsekov commented on DeadlineNetwork/Server/App/Controllers/IParticipantGroupManager.cs:
```diff
@@ -0,0 +1,136 @@
+    public Group Group { get; }
+    public User GroupOwner { get; }
+
+    public ParticipantGroupManager(ApplicationDbContext db, User participant, Group group, User groupOwner)
```
Сделай проверку что `groupOwner` действительно является владельцем группы и если нет то кидай исключение прямо в констуркторе класса

Kemsekov replied:
Так же если `participant` не состоит в группе тоже кидай исключение прямо из конструктора

Kemsekov commented on DeadlineNetwork/Server/App/Controllers/IParticipantGroupManager.cs:
```diff
@@ -0,0 +1,136 @@
+        var task = Db.Tasks.Find(taskId) ?? throw new ArgumentException("Задача не найдена");
+
+        // Проверка доступа на обновление задачи
+        if (task.WhoAdded != Participant.Id && GroupOwner.Id != Participant.Id)
```
можно убрать, если добавишь эту проверку в конструктор

Kemsekov commented on Tests/IParticipantGroupManagerTests.cs:
Нету тестов с исключениями.

Kemsekov reviewed (commented):
доработаь =)(=

teodoroven committed:
Файлы, тесты

Вынес класс ParticipantGroupManager  в отдельный файл ParticipantGroupManager.cs
Переименовал IParticipantGroupManagerTests.cs в ParticipantGroupManagerTests.cs
Написал тесты для всех исключений

teodoroven committed:
Merge branch 'Kemsekov:dev' into dev
